packages/flutter/lib/src/cupertino/picker.dart

```diff
@@ -271,3 +281,105 @@ class _CupertinoPickerState extends State<CupertinoPicker> {
     return result;
   }
 }
+
+class _CupertinoPickerSemantics extends SingleChildRenderObjectWidget {
+  const _CupertinoPickerSemantics({
+    @required this.controller,
+    Widget child,
+  }) : super(child: child);
+
+  final FixedExtentScrollController controller;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderCupertinoPickerSemantics(controller, Directionality.of(context));
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RenderCupertinoPickerSemantics renderObject) {
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..controller = controller;
+  }
+}
+
+class _RenderCupertinoPickerSemantics extends RenderProxyBox {
+  _RenderCupertinoPickerSemantics(FixedExtentScrollController controller, this._textDirection) {
+    this.controller = controller;
+  }
+
+  FixedExtentScrollController get controller => _controller;
+  FixedExtentScrollController _controller;
+  set controller(FixedExtentScrollController value) {
+    if (value == _controller)
+      return;
+    if (_controller != null)
+      _controller.removeListener(_handleScrollUpdate);
+    else
+      _currentIndex = value.initialItem ?? 0;
+    value.addListener(_handleScrollUpdate);
+    _controller = value;
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (textDirection == value)
+      return;
+    _textDirection = value;
+    markNeedsSemanticsUpdate();
+  }
+
+  int _currentIndex = 0;
+
+  void _handleIncrease() {
+    controller.jumpToItem(_currentIndex + 1);
+  }
+
+  void _handleDecrease() {
+    if (_currentIndex == 0)
+      return;
+    controller.jumpToItem(_currentIndex - 1);
```

Comment:
Does this reliably jump to the correct thing?

What if one of the entries in the picker - for whatever reason* - have been excluded from semantics. Wouldn't this jump to the excluded item even though the "decreasedValue" read out by VoiceOver would have been for the next item included in semantics causing a miss-match?

*) one reason I can think of: Maybe "blank" items have been included in the picker for visual grouping? Those most likely do not show up in the semantics tree.

Reply:
Ah! I may have misunderstood what `indexInParent` on SemanticsNode means. It is not the index that a SemanticsNode has has in the child list of its SemanticsNode parent. It's the index that the owning RenderObject of this SemanticsNode had in the child list of its RenderObject parent? Is that correct? If so, that should probably made clearer in the doc to indexInParent, which maybe should also be renamed to something reflecting that parent doesn't refer to the SemanticsNode parent?

Reply:
Also: Maybe instead of having this highly specialized property would it make sense to invent a more general concept of parentData for SemanticsNodes (we already have that for RenderObject) and store the value there? `indexInParent` only seems to make sense for a parent that speaks the same protocol?